### PR TITLE
refactor: centralized router created in file Navigation.tsx

### PR DESCRIPTION
**Pull request by Lil-Noe**
This pull request includes multiple changes across several files to replace the use of the `useRouter` hook from `expo-router` with a new `router` instance from the `App` component. Below are the most important changes:

### Navigation Refactor:
* [`app/App.js`](diffhunk://#diff-927334fc49930ff6860b35d0bc3943fc84d050b4f6b3b3b8e1d15a1e9c90f740R1-R11): Introduced `createNativeStackNavigator` and `useRouter` from `expo-router`, and exported a function that returns the `router` instance.

### Auth Screens:
Replaced `useRouter` with the `router` instance from `App`.

### Other Screens:
 Replaced `useRouter` with the `router` instance from `App`.

### CODE REVIEW

**Comment by Tristan-C789** [comment]
Nice PR! It puts together a really needed navigation system. 

However, I wouldn't have initialized it in `App.js` ; I would use a system similar to the one we used for the Firebase. A `navigation` folder with a `Navigation.tsx` file inside, that basically takes what you put in `App.js`.

`App.js` is a special file for React Native project, and I don't believe it should be used this way, but the concept you had in mind seems good!

**Comment by Tristan-C789** [comment]
It would seem like the APK couldn't be built because there are still some wrong imports in some files. You should fix this before merge!

File: navigation/Navigation.tsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { createNativeStackNavigator } from "@react-navigation/native-stack";
+import { useRouter } from "expo-router";
+
+const Stack = createNativeStackNavigator();
+const router = useRouter();
+
+export default function Nav() {
```
Comment (EduardoNeville) [inline navigation/Navigation.tsx:7]: We should be using a stack navigation from react

File: app/screens/welcome/final_screen.tsx [hunk 2]
```diff
@@ -1,15 +1,16 @@
...
 import React from "react";
 import { StyleSheet, Dimensions } from "react-native";
+import App from "@/app/App";
 
 // Get the screen dimensions
 const { width, height } = Dimensions.get("window");
 
 export default function WelcomeConceptScreen() {
-  const router = useRouter();
+  const router = App();
```
Comment (EduardoNeville) [inline app/screens/welcome/final_screen.tsx:13]: This isn't centralised, we should have our stack navigation in the index.tsx file

File: app/screens/home/home_screen.tsx [hunk 3]
```diff
@@ -5,13 +5,13 @@ import { Challenge } from "@/components/home/Challenge";
 import { ThemedScrollView } from "@/components/theme/ThemedScrollView";
 import { ThemedView } from "@/components/theme/ThemedView";
 import { BottomBar } from "@/components/navigation/BottomBar";
-import { useRouter } from "expo-router";
+import App from "@/app/App";
```
Comment (EduardoNeville) [inline app/screens/home/home_screen.tsx:8]: the same goes for this screen

File: navigation/Navigation.tsx [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { createNativeStackNavigator } from "@react-navigation/native-stack";
```
Comment (EduardoNeville) [inline navigation/Navigation.tsx:1]: we should be using the following tutorial:
https://reactnative.dev/docs/navigation

File: app/screens/camera.tsx [hunk 5]
```diff
@@ -44,7 +44,7 @@ export default function CameraTest() {
   const cameraPictureOptions: CameraPictureOptions = {
     base64: true,
   };
-  const router = useRouter();
+  const router = App();
```
Comment (EduardoNeville) [inline app/screens/camera.tsx:47]: I seek to follow the documentation I have pointed out.

File: app/screens/auth/sign_up_screen.tsx [hunk 6]
```diff
@@ -19,7 +19,7 @@ export default function SignUp() {
   const [email, setEmail] = React.useState("");
   const [password, setPassword] = React.useState("");
   const [confirmPassword, setConfirmPassword] = React.useState("");
-  const router = useRouter();
+  const router = App();
```
Comment (EduardoNeville) [inline app/screens/auth/sign_up_screen.tsx:22]: There are specific use cases for the navigation in the documentation where they handle authentification

File: app/screens/auth/sign_in_screen.tsx [hunk 7]
```diff
@@ -8,18 +8,18 @@ import {
...
 import { ThemedTextButton } from "@/components/theme/ThemedTextButton";
 import { ThemedText } from "@/components/theme/ThemedText";
+import App from "@/navigation/Navigation";
 
 // Get the screen dimensions
 const { width, height } = Dimensions.get("window");
 
 export default function SignInScreen() {
-  const router = useRouter();
+  const router = App();
```
Comment (EduardoNeville) [inline app/screens/auth/sign_in_screen.tsx:22]: Same comment and for the sign_up_screen

**Review by EduardoNeville — DISMISSED** [review]
All in all the effort is good but ultimately, this setup lacks security in terms of the authentification of users when in use of our app.
Secondly, having to call the App components router on each screen seems very far fetched and not very secure.

I would like to point out that we could be using the react navigation system here bellow where we would be able to implement a navigation system that would cover all potential cases. Here is the link:
https://reactnative.dev/docs/navigation

**Review by Tristan-C789 — APPROVED** [review]
It's a much needed PR you made there. There's not much I can say about it, just format it before merging and we'll be good to go!

**Pull request merged**